yt/chyt/controller/internal/strawberry/oplet.go

```diff
@@ -503,6 +533,32 @@ func (oplet *Oplet) needsAbort() (needsAbort bool, reason string) {
 	return false, "up to date"
 }
 
+func (oplet *Oplet) needsSuspend() (needsSuspend bool, reason string) {
```

Comment:
Does it work as expected if we try to suspend already suspended operation, or try to resume not suspended operation?

Reply:
You mean `suspendOp` and `resumeOp` methods? In python sdk suspending a suspended operation does nothing, while resuming a not suspended operation returns an error. I guess both options are acceptable in our case.

Reply:
> You mean `suspendOp` and `resumeOp` methods? In python sdk suspending a suspended operation does nothing, while resuming a not suspended operation returns an error. I guess both options are acceptable in our case.

Yes. It seems to me that if you call the `resume` command on a Jupyter that is not suspended, the `resume_marker` will be changed and the controller will attempt to resume the not-suspended operation indefinitely on every pass.

Reply:
Oh, you are right, let's handle this case in `resumeOp`.